src/libraries/System.Net.NameResolution/tests/FunctionalTests/LoggingTest.cs

```diff
@@ -28,73 +29,95 @@ public static void EventSource_ExistsWithCorrectId()
             Assert.NotEmpty(EventSource.GenerateManifest(esType, "assemblyPathToIncludeInManifest"));
         }
 
-        [ConditionalFact]
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public void GetHostEntry_InvalidHost_LogsError()
         {
-            using (var listener = new TestEventListener("Private.InternalDiagnostics.System.Net.NameResolution", EventLevel.Error))
+            RemoteExecutor.Invoke(static () =>
             {
-                var events = new ConcurrentQueue<EventWrittenEventArgs>();
-
-                listener.RunWithCallback(ev => events.Enqueue(ev), () =>
+                using (var listener = new TestEventListener("Private.InternalDiagnostics.System.Net.NameResolution", EventLevel.Error))
                 {
+                    var events = new ConcurrentQueue<EventWrittenEventArgs>();
+
                     try
                     {
-                        Dns.GetHostEntry(Configuration.Sockets.InvalidHost);
-                        throw new SkipTestException("GetHostEntry should fail but it did not.");
+                        listener.RunWithCallback(ev => events.Enqueue(ev), () =>
+                        {
+                            try
+                            {
+                                Dns.GetHostEntry(Configuration.Sockets.InvalidHost);
+                                throw new SkipTestException("GetHostEntry should fail but it did not.");
+                            }
+                            catch (SocketException e) when (e.SocketErrorCode == SocketError.HostNotFound)
+                            {
+                            }
+                            catch (Exception e)
+                            {
+                                throw new SkipTestException($"GetHostEntry failed unexpectedly: {e.Message}");
+                            }
+                        });
                     }
-                    catch (SocketException e) when (e.SocketErrorCode == SocketError.HostNotFound)
+                    catch (SkipTestException)
                     {
+                        // Can't throw SkipTestException here because we're running in RemoteExecutor.
```

Comment:
Could we instead return a specific error code that the caller than translates into SkipTestException?